subprojects/docs/src/samples/credentials-handling/publishing-credentials/tests/publishCommandLineCredentials.sample.conf

```diff
@@ -0,0 +1,4 @@
+commands: [{
```

Comment:
I haven't tried this locally yet, but I think something like this might work:

- Move MavenRepositoryStub into an included build as a java application instead of buildSrc
- As steps in this test:
   - install the maven repository app 
   - Run the maven repository app (in the background)
   - Run the Gradle build to publish to the app
   - Stop the maven repository app

I think that keeps the weirdness of having to start/stop a maven repository out of the real build.  The difficult parts will be running the application in the background, knowing which port to use and stopping the application.

Reply:
I've been going back and forth with this for a bit - while this is doable, I think it is the difficult parts you mentioned that make it even more awkward/complex in the end.
Tracking the lifecycle of the app started in the sample is tricky - I was thinking to either timeout after a period (quick and dirty) or otherwise record the PID somehow to kill the server in the last step.
Since the port is a random free one - and I think having a hardcoded one instead would not be a good idea - the only way to pass it between test steps I could think of is to write it to a `gradle.properties` that's consumed by the publishing build.
If this flow is made to work, I think it also breaks the user experience where we want the sample to be runnable.

In the end, I managed to encapsulate the whole server stubbing in a precompiled plugin (included build) - the only invasive bit in the main code remains the `id 'maven-repository-stub'` plugin application.